src/frontends/tensorflow/src/graph_iterator_proto.hpp

```diff
@@ -50,12 +91,27 @@ class GraphIteratorProto : public GraphIterator {
     }
 
     bool is_end() const override {
-        return node_index >= m_nodes.size();
+        return node_index >= m_decoders.size();
     }
 
     /// Return NodeContext for the current node that iterator points to
     std::shared_ptr<DecoderBase> get_decoder() const override {
-        return std::make_shared<DecoderProto>(m_nodes[node_index]);
+        return m_decoders[node_index];
+    }
+
+    std::shared_ptr<GraphIterator> get_body_graph_iterator(const std::string& func_name) const override {
+        // check that the requested function exists in the library
+        auto num_funcs = m_graph_def->library().function_size();
+        for (int func_ind = 0; func_ind < num_funcs; ++func_ind) {
```

Comment:
This will work inefficiently for models with many functions. You will search linearly every time. Consider to cache the function list in constructor.

Reply:
Good point, but we don't have a lot of functions in the library in real models. I never saw more that 5 functions.:) Not critical  I think.

Reply:
@mvafin, @itikhono, I will fix it. I would rather to create a map<string, int> to save index by name to avoid extra memory consumption for unneeded functions.

Reply:
Fixed.